### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by jmberg**
Just a few cleanups of things I came across while looking at libsml for my DZG meter

### CODE REVIEW

**Comment by jmberg** [comment]
FWIW, the DZG meter workaround I implemented merges on top of this - if you don't like these changes, I can rebase, but I won't until you reject this.

**Comment by sir106** [comment]
tested locally; cleanup works fine.

**Comment by andig** [comment]
Out of curiosity- why is the __func__ change necessary?

**Comment by sir106** [comment]
As I understood it is to get rid of the warning messages at compile time.

__FUNCTION__ is deprecated, and GCC 3.2 will handle __FUNCTION__ the same way as __func__. 
__func__ is defined by the ISO standard C99:

__func__ was added to C++ in C++11, where it is specified as containing "an implementation-deﬁned string" (C++11 §8.4.1[dcl.fct.def.general]/8), which is not quite as useful as the specification in C. (The original proposal to add __func__ to C++ was N1642).

__FUNCTION__ is a pre-standard extension that some C compilers support (including gcc and Visual C++); in general, you should use __func__ where it is supported and only use __FUNCTION__ if you are using a compiler that does not support it (for example, Visual C++, which does not support C99 and does not yet support all of C++0x, does not provide __func__).

File: sml/src/sml_number.c [hunk 1]
```diff
@@ -24,8 +24,25 @@
 #define SML_BIG_ENDIAN 1
 #define SML_LITTLE_ENDIAN 0
 
-int sml_number_endian();
-void sml_number_byte_swap(unsigned char *bytes, int bytes_len);
+static void sml_number_byte_swap(unsigned char *bytes, int bytes_len) {
+	int i;
+	unsigned char ob[bytes_len];
```
Comment (mbehr1) [inline sml/src/sml_number.c]: that's a bit risky as bytes_len can be huge. (but it was part of orig. change already)

File: sml/src/sml_number.c [hunk 2]
```diff
@@ -24,8 +24,25 @@
...
+
+	memcpy(&ob, bytes, bytes_len);
+
+	for (i = 0; i < bytes_len; i++)
+		bytes[i] = ob[bytes_len - (i + 1)];
+}
+
+static int sml_number_endian() {
```
Comment (mbehr1) [inline sml/src/sml_number.c]: typically you don't need to know that but e.g. use from endian.h functions like be32toh...

File: sml/src/sml_tree.c [hunk 3]
```diff
@@ -289,7 +289,7 @@ void sml_proc_par_value_write(sml_proc_par_value *value, sml_buffer *buf) {
 			sml_time_write(value->data.time, buf);
 			break;
 		default:
-			fprintf(stderr,"libsml: error: unknown tag in %s\n", __FUNCTION__);
+			fprintf(stderr,"libsml: error: unknown tag in %s\n", __func__);
```
Comment (mbehr1) [inline sml/src/sml_tree.c]: would be nice to print the unknown tag as well

**Comment by andig** [comment]
@sir106 would you be interested in updating with the endian changes suggested by mbehr?

File: sml/src/sml_tree.c (hunk 3 above)
Comment (andig) [reply sml/src/sml_tree.c]: Please add this, too

**Comment by jmberg** [comment]
I'd argue the changes should be a separate commit (and then there's no point in adding it to this pull request), but you're free to manage the project as you wish :-)

**Comment by andig** [comment]
Thank you. Would be great if you want to help with the additional improvements.

**Comment by jmberg** [comment]
I note that the patches in this pull request were all squashed into one that has no details whatsoever - I don't know what the github UI offers you, but according to https://help.github.com/articles/merging-a-pull-request/ you have different options. I'd argue that for a multi-patch pull request you should never use the "squash" option as it deletes the history of the changes.

Just my 2 cents.

**Comment by andig** [comment]
> I'd argue that for a multi-patch pull request you should never use the "squash" option as it deletes the history of the changes.

Depends on the quality of the commits. Seem this time I should have merged, sorry about that.

**Comment by jmberg** [comment]
Fair enough, personally (and in my Linux maintainer role) I'd just reject bad commits instead and tell people to go rework them :-)

**Pull request merged**